macos mixed C/C++ and Zig builds fail

The last working zig version was 0.7.0 for mixed builds on macos. There have been various different errors since then but it seems the error has now stabilized so I figure I'll report it.

The test project is available [here](https://github.com/prime31/zig-upaya). Full steps to repro:
- build zig from master (tested every few days since 0.7.0 dropped)
- `git clone --recursive https://github.com/prime31/zig-upaya/`
- `cd zig-upaya`
- `zig build empty`

The resulting error seems to be a missing include path that has popped up after 0.7.0:
```
Compile C Objects [1/9] cimgui.cpp... error(compilation): clang failed with stderr: In file included from /Users/mikedesaro/zig-upaya/src/deps/sokol/compile_sokol.c:12:
/Users/mikedesaro/zig-upaya/src/deps/sokol/sokol/sokol_app.h:1088:17: fatal error: 'Metal/Metal.h' file not found
```

--- a/src/main.zig
+++ b/src/main.zig
@@ -1432,6 +1432,12 @@ fn buildOutputType(
         }
     }
 
+    if (comptime std.Target.current.isDarwin()) {
+        // If we want to link against frameworks, we need system headers.
+        if (framework_dirs.items.len > 0 or frameworks.items.len > 0)
+            want_native_include_dirs = true;
+    }
+
     if (cross_target.isNativeOs() and (system_libs.items.len != 0 or want_native_include_dirs)) {
         const paths = std.zig.system.NativePaths.detect(arena) catch |err| {
             fatal("unable to detect native system paths: {s}", .{@errorName(err)});

